src/helpers.rs

```diff
@@ -348,4 +349,42 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         Ok(())
     }
+
+    fn read_os_string(&mut self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString> {
+        let bytes = self.eval_context_mut().memory().read_c_str(scalar)?.to_vec();
+        if cfg!(unix) {
+            Ok(std::os::unix::ffi::OsStringExt::from_vec(bytes))
+        } else {
+            std::str::from_utf8(&bytes)
```

Comment:
This should come with a comment explaining that on non-Unix platforms, we use the only generically (cross-platform) available way of creating an `OsString`, which is from a `String`. This rules out some otherwise valid paths (such as non-UTF-8 paths), but it is the best we can do.